### ISSUE TITLE
False  positive for E206

### ISSUE THREAD

**Issue by lisael** [ansible/ansible-lint#505]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.7.8

ansible-lint --version
ansible-lint 4.1.0
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

206 should not be reported on non-jinja tags, e.g inside jinja strings:

```yaml
- name: Get mycontainer's PID
  command: "docker inspect --format={{ '{{.State.Pid}}' }} mycontainer"
  register: mycontainers_pid
```
`{{.State.Pid}}` is a go format tag, not a jinja one

# Actual Behaviour (Bug report only)

```
[206] Variables should have spaces before and after: {{ var_name }}
/home/lisael/myplaybook.yml:21
  command: "docker inspect --format={{ '{{.State.Pid}}' }} mycontainer"
```

**Comment by ssbarnea** [comment]
I am facing the same bug on other plain simple strings:
```
variable_start_string: "{{"
variable_end_string: "}}"
```
Does anyone knows a workaround for these false positives?

**Comment by awcrosby** [comment]
@ssbarnea the particular case in your examples should be taken care of by #535, we plan for a new bugfix release at the end of this month.

**Comment by ssbarnea** [comment]
@awcrosby Thanks. Meanwhile I was able to workaround it by using jinja2 documentation: ``{{ '{{' }}`` .. ugly but valid.

**Comment by stdedos** [comment]
@awcrosby `${ab.c.${d}}` is also a false-positive

**Comment by ssbarnea** [comment]
And here is a new false positive that is related to jinja2 and I have no idea how to avoid that one:
```
- set_fact:
    host_addresses: >
      {% set hosts = {} -%}
      {% for host, vars in hostvars.items() -%}
      {% set _ = hosts.update({host: vars['nodepool']['private_ipv4']}) -%}
      {% set _ = hosts.update({vars['ansible_hostname']: vars['nodepool']['private_ipv4']}) -%}
      {% endfor -%}
      {{- hosts -}}
```
It chokes on the last line because it considers that `-` is part of the variable name and having it unspaces is key for having jinja2 working.

**Comment by dayja78** [comment]
My apologies for being late to the party.  Looks like my particular false-positive case was already documented in #499 and fixed in #500.

**Comment by axeliux** [comment]
I worked around if by scaling it

{% row %}
{% endraw %}

**Comment by webknjaz** [comment]
If anybody can send a pytest-based test reproducing this as a PR it'd be great.

**Comment by mickaeltr** [comment]
Hello,

I am hitting the same problem with:

```yaml
log_options: "tag={% raw %}{{.ImageName}}{% endraw %}"
log_options: !unsafe "tag={{.ImageName}}"
```

**Comment by ssbarnea** [comment]
Use `# noqa` to skip these false positives or raise a PR that improves the regex match.

**Comment by stdedos** [comment]
> I tried to use a temporary workaround of adding `# noqa 206` but that is not possible due to #557

If that is still true, then how can we skip multiline false positives? 😕

**Comment by ssbarnea** [comment]
You can put the noqa comment on the name line, first line of the task. In fact that is the line number reported for task errors: the first line of the task.

```yaml
    - name: foo  # noqa 305
      shell: >-
        /usr/bin/true
        --help
```

**Comment by lisael** [comment]
@ssbarnea Regarding my original issue, ( `{{xyz` in a jinja string) I believe it's hard to fix the regex, and I'm not sure we want a full-blown parser for this. I'll go with the `#noqa` workaround where possible. Thank you.
